### PR TITLE
Refactor directive prologue tests for parsers

### PR DESCRIPTION
**Pull request by jugglinmike**
@leobalter @rwaldron This is in service of gh-1356. The tests modified in this patch have a trait that differentiates them from the tests we've previously reformatted for the benefit of parsers. Because they concern the observance of strict mode, there are parsing *and* runtime behaviors that we need to verify.  So while in those previous patches, we could just replace the calls to `eval` with direct program code, in this case, I've taken care to preserve assertions for runtime semantics. That said, I've still modified those tests in order to make this consideration more clear (they now test for strict mode through assignment instead of `eval`).

To facilitate review (and help verify that I didn't invalidate any of the old tests), I've split this work into a number of distinct commits. Hope that helps!

### CODE REVIEW

File: test/language/directive-prologue/func-decl-no-semi-parse.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2018 Mike Pennisi.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-2-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-decl-no-semi-parse.js]: Add or replace es5id with: 

```
esid: use-strict-directive
```

File: test/language/directive-prologue/func-decl-inside-func-decl-parse.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2018 Mike Pennisi.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-15-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-decl-inside-func-decl-parse.js]: Add or replace es5id with: 

```
esid: sec-strict-mode-code
```

File: test/language/directive-prologue/func-decl-no-semi-runtime.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2012 Ecma International.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-2-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-decl-no-semi-runtime.js]: Add or replace es5id with: 

```
esid: use-strict-directive
```

Comment (rwaldron) [inline test/language/directive-prologue/func-decl-no-semi-runtime.js]: Add or replace es5id with: 

```
esid: use-strict-directive
```

File: test/language/directive-prologue/func-decl-parse.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2018 Mike Pennisi.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-19-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-decl-parse.js]: Add or replace es5id with: 

```
esid: sec-strict-mode-code
```

File: test/language/directive-prologue/func-expr-inside-func-decl-parse.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2018 Mike Pennisi.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-16-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-expr-inside-func-decl-parse.js]: Add or replace es5id with: 

```
esid: sec-strict-mode-code
```

File: test/language/directive-prologue/func-expr-no-semi-parse.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2018 Mike Pennisi.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-2-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-expr-no-semi-parse.js]: Add or replace es5id with: 

```
esid: sec-strict-mode-code
```

File: test/language/directive-prologue/func-expr-parse.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2018 Mike Pennisi.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-22-s
```
Comment (rwaldron) [inline test/language/directive-prologue/func-expr-parse.js]: Add or replace es5id with: 

```
esid: sec-strict-mode-code
```

File: test/language/directive-prologue/set-accsr-runtime.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2012 Ecma International.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es5id: 10.1.1-25-s
```
Comment (rwaldron) [inline test/language/directive-prologue/set-accsr-runtime.js]: Add or replace es5id with: 

```
esid: sec-strict-mode-code
```

**Comment by jugglinmike** [comment]
@rwaldron Metadata added!

**Comment by jugglinmike** [comment]
It looks like the linter has a faulty rule. There are a number of ways we could address this:

- add exceptions for the new files to the linter's white list
- extend the regular expression with a special case for this particular `esid` value
- remove the linting rule
- relax the linting rule to simply assert the value of `esid` satisfies the general requirements for HTML document fragment identifiers
- extend the linter to fetch the spec and scrape out the value identifiers

I'm happy to help out with any of these, though my preference would be the fourth. @rwaldron what do you think?

**Comment by rwaldron** [comment]
https://github.com/jugglinmike/test262/pull/2

File: test/language/directive-prologue/func-decl-final-runtime.js [hunk 9]
```diff
@@ -2,15 +2,16 @@
...
     Strict Mode - Function code of a FunctionDeclaration contains Use
     Strict Directive which appears at the end of the block
 flags: [noStrict]
 ---*/
         function fun() {
-            eval("var public = 1;");
-            assert.sameValue(public, 1);
+            test262unresolvable = null;
+            assert.sameValue(test262unresolvable, null);
             "use strict";
```
Comment (leobalter) [inline test/language/directive-prologue/func-decl-final-runtime.js]: I wonder if we have some coverage for the pragma only preceded by an empty statement. I did not find anything from a quick grep.

```js
function fun() {
  ;'use strict';
  unr = null;
  assert.sameValue(unr, null);
}
fun();
```

**Comment by leobalter** [comment]
> relax the linting rule to simply assert the value of esid satisfies the general requirements for HTML document fragment identifiers

I'm positive with this solution.

**Comment by jugglinmike** [comment]
To unblock this, I've submitted https://github.com/tc39/test262/pull/1632 (as an alternative to https://github.com/jugglinmike/test262/pull/2).

**Comment by rwaldron** [comment]
I've rebased this locally and confirmed that linting passes: 

```
rwaldron in ~/clonez/test262 on refactor-for-parsers-directive
$ ./tools/lint/lint.py --whitelist lint.whitelist test/language/directive-prologue/
Linting 62 files
Linting complete. 0 errors found.
```

**Comment by rwaldron** [comment]
Merged locally

**Pull request closed without merging**